File: src/lib/stores/goals.ts
import { writable } from 'svelte/store';
import { invoke } from '$lib/tauri';

export interface Goal {
  id: string;
  name: string;
  locked_until: string;
}

export const goals = writable<Goal[]>([]);

export async function loadGoals() {
  const result = await invoke<Goal[]>('cmd_list_goals');
  goals.set(result);
}

export async function addGoal(name: string) {
  await invoke('cmd_create_goal', { name });
  await loadGoals();
}

File: src-tauri/src/commands/goals.rs
use crate::{goals, AppState};
use tauri::State;

#[tauri::command]
pub async fn cmd_create_goal(
    state: State<'_, AppState>,
    name: String,
) -> Result<goals::Goal, String> {
    goals::create_goal(&state.db, &name, &state.device_id).await
}

#[tauri::command]
pub async fn cmd_list_goals(state: State<'_, AppState>) -> Result<Vec<goals::Goal>, String> {
    goals::list_goals(&state.db).await
}
